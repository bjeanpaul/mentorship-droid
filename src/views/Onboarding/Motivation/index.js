import React, { PropTypes } from 'react';

import { MultiLineTextInput, FormStep } from 'src/components';


const Motivation = ({
  profile: { motivation = '' },
  onChange,
  ...props,
}) => (
  <FormStep
    title="Why did you become a mentor?"
    paginationDisabled={motivation.length === 0}
    {...props}
  >
    <MultiLineTextInput
      value={motivation}
      placeholder="Type your answer here"
      maxLength={50}
      onChangeText={text => onChange({ motivation: text })}
    />
  </FormStep>
);

Motivation.propTypes = {
  profile: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
  onNextPress: PropTypes.func.isRequired,
};


export default Motivation;
