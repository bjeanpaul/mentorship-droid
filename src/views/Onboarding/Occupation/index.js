import React, { PropTypes } from 'react';

import { TextInput } from 'src/components';
import FormStep from 'src/containers/OnboardingFormStepContainer';


const Occupation = ({
  jobTitle = '',
  jobSector = '',
  onChangeText,
}) => (
  <FormStep
    paginationDisabled={jobTitle.length === 0 || jobSector.length === 0}
    title="What do you do?"
  >
    <TextInput
      label="Job Sector"
      value={jobSector}
      onChangeText={text => onChangeText({ jobSector: text })}
    />
    <TextInput
      label="Job Title"
      value={jobTitle}
      onChangeText={text => onChangeText({ jobTitle: text })}
    />
  </FormStep>
);

Occupation.propTypes = {
  jobSector: PropTypes.string,
  jobTitle: PropTypes.string,
  onChangeText: PropTypes.func.isRequired,
};


export default Occupation;
