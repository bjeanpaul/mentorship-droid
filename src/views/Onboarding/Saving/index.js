import React, { PropTypes } from 'react';
import { OverlayLoading } from 'src/components';


class Saving extends React.Component {
  // TODO: Stepper should dispatch an action when it's on the last view
  componentDidMount() {
    const { profile } = this.props;
    this.props.save(profile);
  }

  render() {
    return <OverlayLoading title="Saving profile..." />;
  }
}


Saving.propTypes = {
  profile: PropTypes.any.isRequired,
  save: PropTypes.func.isRequired,
};


export default Saving;
