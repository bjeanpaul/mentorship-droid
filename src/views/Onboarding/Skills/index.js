import React, { PropTypes } from 'react';

import { MultiLineTextInput, FormStep } from 'src/components';


const Skills = ({
  profile,
  onChange,
  onDonePress,
  ...props,
}) => {
  const { skills = '' } = profile;

  return (
    <FormStep
      last
      title="What skills and strengths do you have that will help you as a Mentor?"
      paginationDisabled={skills.length === 0}
      onDonePress={() => onDonePress(profile.id, profile)}
      {...props}
    >
      <MultiLineTextInput
        value={skills}
        placeholder="Type your answer here"
        maxLength={50}
        onChangeText={text => onChange({ skills: text })}
      />
    </FormStep>
  );
};


Skills.propTypes = {
  profile: PropTypes.object,
  onChange: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
  onDonePress: PropTypes.func.isRequired,
};


export default Skills;
