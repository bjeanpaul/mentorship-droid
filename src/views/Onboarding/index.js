import React, { PropTypes } from 'react';
import { Stepper } from 'src/components';
import containers from 'src/containers/OnboardingStepsContainer';


const Onboarding = ({
  navigationState,
}) => (
  <Stepper navigationState={navigationState}>
    <containers.Greeting />
    <containers.ProfilePicture />
    <containers.Occupation />
    <containers.Inspiration />
    <containers.Motivation />
    <containers.ThreeWords />
    <containers.Skills />
    <containers.Saving />
  </Stepper>
);


Onboarding.propTypes = {
  navigationState: PropTypes.object.isRequired,
};


export default Onboarding;
