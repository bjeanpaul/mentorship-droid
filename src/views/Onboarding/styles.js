import { StyleSheet } from 'react-native';
import colors from 'src/constants/colors';

export default StyleSheet.create({
  hint: {
    textAlign: 'left',
    marginLeft: 24,
    marginRight: 24,
    color: colors.TEXT_INPUT_HINT,
    fontSize: 14,
  },
});
