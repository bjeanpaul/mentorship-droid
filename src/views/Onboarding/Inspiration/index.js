import React, { PropTypes } from 'react';

import { Text, MultiLineTextInput, FormStep } from 'src/components';
import styles from 'src/views/Onboarding/styles';


const Inspiration = ({
  profile: { inspiration = '' },
  onChange,
  ...props,
}) => (
  <FormStep
    title="Who would you love as your own mentor?"
    paginationDisabled={inspiration.length === 0}
    {...props}
  >
    <MultiLineTextInput
      value={inspiration}
      placeholder="Type your answer here"
      onChangeText={text => onChange({ inspiration: text })}
    />
    <Text style={styles.hint}>e.g. Malala, Sania Nehwal, Chanda Kocchar</Text>
  </FormStep>
);

Inspiration.propTypes = {
  profile: PropTypes.object,
  onChange: PropTypes.func.isRequired,
  onBackPress: PropTypes.func,
  onNextPress: PropTypes.func,
};


export default Inspiration;
