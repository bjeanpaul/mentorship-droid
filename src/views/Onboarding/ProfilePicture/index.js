import React, { PropTypes } from 'react';
import { View, Image, TouchableWithoutFeedback } from 'react-native';

import { Text, FormStep } from 'src/components';
import images from 'src/constants/images';
import styles from './styles';


const ProfilePicture = ({
  profile: { profilePic },
  onChoosePhotoPress,
  ...props,
}) => (
  <FormStep
    title="Add a profile picture"
    paginationDisabled={!profilePic}
    {...props}
  >
    <TouchableWithoutFeedback onPress={onChoosePhotoPress}>
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={
            profilePic
              ? { uri: profilePic }
              : images.PROFILE_FACE_ADD
          }
        />
        <Text style={styles.imageLabel}>Choose Photo</Text>
      </View>
    </TouchableWithoutFeedback>
  </FormStep>
);


ProfilePicture.propTypes = {
  profile: PropTypes.object.isRequired,
  onChoosePhotoPress: PropTypes.func.isRequired,
  onBackPress: PropTypes.func.isRequired,
  onNextPress: PropTypes.func.isRequired,
};


export default ProfilePicture;
