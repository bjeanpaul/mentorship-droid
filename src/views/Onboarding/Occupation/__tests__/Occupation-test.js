import { noop } from 'lodash';
import React from 'react';

import { fakeProfile } from 'app/scripts/helpers';
import { Occupation } from 'src/views/Onboarding';


describe('Occupation', () => {
  const createComponent = (props = {}) => (
    <Occupation
      profile={fakeProfile({
        jobTitle: 'Uncle',
        jobSector: 'Family',
      })}
      onChange={noop}
      onBackPress={noop}
      onNextPress={noop}
      {...props}
    />
  );

  it('should render', () => {
    const el = render(createComponent());
    expect(el.toJSON()).toMatchSnapshot();
  });
});

