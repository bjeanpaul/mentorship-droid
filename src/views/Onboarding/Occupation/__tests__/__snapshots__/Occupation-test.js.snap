exports[`Occupation should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center",
        },
        Array [
          Object {
            "fontFamily": "Brandon Text_medium",
            "fontSize": 20,
            "fontWeight": "400",
          },
          Object {
            "margin": 25,
            "marginBottom": 50,
          },
        ],
      ]
    }>
    What do you do?
  </Text>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#dfe5e6",
          "marginBottom": 27,
          "marginLeft": 24,
          "marginRight": 24,
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#9fb1b3",
              "fontFamily": "Brandon Text_bold",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 9,
              "textAlign": "left",
            },
            undefined,
          ],
        ]
      }>
      Job Sector
    </Text>
    <TextInput
      label="Job Sector"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#c0cbcc"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 18,
            "lineHeight": 28,
            "marginBottom": 8,
            "padding": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="Family" />
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#dfe5e6",
          "marginBottom": 27,
          "marginLeft": 24,
          "marginRight": 24,
        },
        false,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#9fb1b3",
              "fontFamily": "Brandon Text_bold",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 9,
              "textAlign": "left",
            },
            undefined,
          ],
        ]
      }>
      Job Title
    </Text>
    <TextInput
      label="Job Title"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#c0cbcc"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
            "fontSize": 18,
            "lineHeight": 28,
            "marginBottom": 8,
            "padding": 0,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value="Uncle" />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }>
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 16,
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }>
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }>
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
